ocfweb/docs/docs/staff/backend/internal-firewalls.md

```diff
@@ -0,0 +1,126 @@
+   found in the puppet code. Generally:
+    * DMZ servers can talk to needed production services on the appropriate
+      ports.
+    * Staff VMs allow all incoming traffic.
+    * Staff VMs and desktops are allowed to directly SSH into internal servers.
+    * There are other miscellaneous rules allowing other traffic.
+
+In addition to input rules, it's necessary to have some output rules as well to
+protect devices that we don't trust to have reliable firewalls, such as printers
+and IPMI devices. The rules for those devices work a little differently:
+
+ * A handful of "uber-trusted" servers, such as supernova and hypervisors, are
+   allowed to talk to all of these special devices. This allows us to use these
+   servers for the purposes of configuring or debugging those devices.
+ * Some other servers can talk to certain special devices. For example, the
+   print server can talk to printers.
+ * Most other hosts cannot talk to these special devices.
+
+Output rules are not a perfect solution, since they operate on a voluntary
+mechanism and can't prevent non-OCF hosts that may be connected to our network
+from contacting these special devices anyway. Preventing this is a future
+project.
+
+It's important to note that internal firewalls are set up to only filter traffic
+from other OCF machines. Traffic from outside the OCF network is the sole
+responsibility of the external firewall.
+
+[dmz]: https://en.wikipedia.org/wiki/DMZ_(computing)
+
+
+## Implementation
+
+Internal firewalls are implemented using iptables rules set by Puppet with the
+[puppetlabs-firewall module][puppetlabs-firewall]. We place all of our input
+rules in the `PUPPET-INPUT` chain and all of our output rules in the
+`PUPPET-OUTPUT` chain.
+
+Firewall rules are added by using `firewall_multi` and
+`ocf::firewall::firewall46` declarations:
+ * `ocf::firewall::firewall46` should generally be used in most cases. It
+   inserts IPv4 and IPv6 iptables rules, but only adds the IPv6 iptables rule if
+   the host has a public IPv6 address. This prevents Puppet errors otherwise
+   occurring due to IPv6 addresses not being resolved.
+ * `firewall_multi` should be used if IP addresses need to be manually specified
+   in the firewall rule.
+ * `ocf::firewall::firewall46` and `firewall_multi` both internally use the
+   `firewall` resource. Direct use of the `firewall` resource should be avoided
+   since such resources wouldn't be subject to the [ordering constraints
+   generally placed on firewall resources][ordering].
+
+[ordering]: https://github.com/ocf/puppet/blob/f3fdd5912a5dc5eafd9995412a9c5e85874dee31/manifests/site.pp#L50-L58
+[puppetlabs-firewall]: https://forge.puppet.com/puppetlabs/firewall
+
+
+## Debugging
+
+### `iptables` commands
+
+Note that all of the following commands need to be run as root.
+
+The `iptables` command allows you to inspect and debug IPv4 firewall rules:
+
+ * `iptables -L`: list firewall rules
```

Comment:
`iptables-save` is a useful command (and usually more useful, imo) for inspecting iptables state

Reply:
Thanks for the suggestion; I've added `iptables -S`, which is a superset.